pkg/gatewayserver/io/io.go

```diff
@@ -90,6 +90,15 @@ type Connection struct {
 
 	statsChangedCh chan struct{}
 	locCh          chan struct{}
+
+	newTraffic Traffic
```

Comment:
I really don't think we should be keeping state like this

Reply:
Changed to channels per traffic type.